server/util/lib/set/set.go

```diff
@@ -0,0 +1,145 @@
+package set
+
+import (
+	"iter"
+	"maps"
+)
+
+// View is a (read-only) view of a data structure as a set.
+type View[E comparable] interface {
+	All() iter.Seq[E]
+	Contains(E) bool
+	Intersection(View[E]) iter.Seq[E]
+	Union(View[E]) iter.Seq[E]
+	Difference(View[E]) iter.Seq[E]
+}
+
+// Nothing is an alias for struct{} for clarity/readability.
+type Nothing = struct{}
+
+// Set is just a lightweight wrapper around the standard golang stand-in for the
+// set type, map[E]struct{}. It is intended to improve readability and reduce
+// code duplication. An empty Set can be made with `make(Set[E])` or
+// `make(Set[E], cap)`, just like a normal map.
+type Set[E comparable] map[E]Nothing
+
+// Set is just a lightweight wrapper around the standard golang stand-in for the
+// set type, map[E]struct{}. It is intended to improve readability and reduce
+// code duplication. An empty Set can be made with `make(Set[E])` or
+// `make(Set[E], cap)`, just like a normal map.
```

Comment:
this comment needs to be fixed; it was obviously copy-pasted from above.

Reply:
```suggestion
// Set is just a lightweight wrapper around the standard golang stand-in for the
// set type, map[E]struct{}. It is intended to improve readability and reduce
// code duplication. An empty Set can be made with `make(Set[E])` or
// `make(Set[E], cap)`, just like a normal map.
```

// mapView is a view of a `map[E]V` as a View[E]